modules/autostash.lua

```diff
@@ -442,6 +456,126 @@ local function insert_item_into_chest(stack, chests, filtered_chests, name, floa
     end
 end
 
+local function collect_from(sources, player_inventory, quick_bar_request, floaty_text_list)
+    for _, source in pairs(sources) do
+        if table.size(quick_bar_request) == 0 then
+            return
+        end
+
+        if not source.inventory then
+            goto next_chest
+        end
+
+        local source_contents = source.inventory.get_contents()
+
+        for name, request_count in pairs(quick_bar_request) do
+            local chest_count = source_contents[name]
+            if chest_count == nil or chest_count == 0 then
+                goto next_item
+            end
+
+            local count_to_insert = math.min(chest_count, request_count)
+            local actually_inserted = player_inventory.insert({ name = name, count = count_to_insert })
```

Comment:
This code is creating new items and deletes old items in the target container, that is fine with basic items, but with any items with additional data (durability, ammo amount, modules in armors, and so on) this deletes the data. As far as I know, it is possible (and also viable strategy) to have even armors on quick bar, this would instantly delete the insides of fetched items. Also it would reset the science packs durability to full, and similar.

If you check the code that inserts the items from player to chests, we work with actual item stacks (specific slot of items) that are transported, as that transports also the additional data. And then we modify the stack sizes. The basic insert/delete can be done only with basic items like ores or plates, and as far as I see, this function doesn't filter the item types to deal with this.

Reply:
Ok, thanks, I'll check or maybe just will use blacklist of prototypes with additional data.